### ISSUE TITLE
sql: column spuriously considered to be in multiple FKs

### ISSUE THREAD

**Issue by dt** [cockroachdb/cockroach#8345]
currently all the columns in an index that is being used for an FK are considered to be participating and thus not allowed to participate in any other FKs, but when only the strict prefix of an index is being used, some columns that are not actually participating may become incorrectly banned form participating in another FK.

``````
CREATE TABLE foo (a INT PRIMARY KEY, b INT UNIQUE);

CREATE TABLE bar (
a INT REFERENCES foo (a),
b INT REFERENCES foo (b),
INDEX (a, b),
INDEX (b)
)```
``````

**Comment by nvanbenschoten** [comment]
Another way that this issue can materialize is when a foreign key is using a prefix of a table's primary key. An example is:

```
root@:26257> CREATE TABLE customers (id INT PRIMARY KEY);
CREATE TABLE
root@:26257> CREATE TABLE orders (
                                   customer INT, 
                                   id INT, 
                                   PRIMARY KEY (customer, id), 
                                   CONSTRAINT fk_customer FOREIGN KEY (customer) REFERENCES customers
                                 );
CREATE TABLE
root@:26257> SHOW CONSTRAINTS FROM orders;
+--------+-------------+-------------+--------------+----------------+
| Table  |    Name     |    Type     |  Column(s)   |    Details     |
+--------+-------------+-------------+--------------+----------------+
| orders | fk_customer | FOREIGN KEY | customer, id | customers.[id] |
| orders | primary     | PRIMARY KEY | customer, id | NULL           |
+--------+-------------+-------------+--------------+----------------+
(2 rows)
```
(notice `customer, id` in the `Column(s)` column for the first row)

**Comment by dt** [comment]
I initially though we'd wait to fix this along with the matched-column-order-in-index requirement, but maybe a quick fix like #13439 could serve in the interim

**Comment by dt** [comment]
fixed in #13439
